src/main/java/pianokeys/PianoController.java

```diff
@@ -34,7 +51,19 @@ public void playNote(int note)
         pianoView.showKeyPlayed(note, true);
         if (recording)
         {
-            recorder.startNote(note);
+            activeNotes.add(note);
+            double startTime = compositionView.getCurrentTime();
+            recorder.startNote(note, startTime);
+
+            heldNoteStartTimes.put(note, startTime);
+            heldNoteStartTimestamps.put(note, System.currentTimeMillis());
+
+            if (!repaintTimer.isRunning())
+            {
+                repaintTimer.start();
+            }
+
+            compositionView.repaint();
```

Comment:
This does not need to be called immediately since the repaintTimer will call it in 50ms